speechbrain/k2_integration/lattice_decoder.py

```diff
@@ -0,0 +1,339 @@
+"""Different decoding graph algorithms for k2, be it HL or HLG (with G LM
+and bigger rescoring LM).
+
+This code was adjusted from icefall (https://github.com/k2-fsa/icefall/blob/master/icefall/decode.py).
+
+
+Authors:
+  * Pierre Champion 2023
+  * Zeyu Zhao 2023
+  * Georgios Karakasidis 2023
+"""
+
+from pathlib import Path
+from typing import Dict, List, Optional, Union
+from collections import OrderedDict
+
+from . import k2 # import k2 from ./__init__.py
+from speechbrain.utils.distributed import run_on_main
+
+import torch
+import logging
+
+from . import graph_compiler, utils
+
+logger = logging.getLogger(__name__)
+
+def get_decoding(
+    hparams,
+    graphCompiler: graph_compiler.GraphCompiler,
```

Comment:
I prefer explicit relative imports a lot more, just by glancing at the arg, you know where it comes from instead of remembering all of the possible import of k2_integration.